File: app/src/settings/infrastructure/stable.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import { ref } from 'vue';
import { defineStore } from 'pinia';

export const useStableSettings = defineStore('stable', () => {
  const hideUnmet = ref(true);

  return {
    hideUnmet,
  };
});

File: crates/nil-payload/src/response/infrastructure/stable.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::infrastructure::building::stable::recruit_catalog::StableRecruitCatalog;
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[cfg(feature = "axum")]
use nil_payload_macros::IntoJsonResponse;

#[derive(Clone, Debug, Deserialize, Serialize, TS)]
#[cfg_attr(feature = "axum", derive(IntoJsonResponse))]
#[ts(export)]
pub struct GetStableRecruitCatalogResponse(pub StableRecruitCatalog);
